File: Ruby/P3/Irrgarten/game_state.rb
#encoding:utf-8


module Irrgarten
    class GameState

        # Constructor 

        def initialize (labyrinth, players, monsters, current_player,
                        winner, log)
            @labyrinth = labyrinth 
            @players = players 
            @monsters = monsters 
            @current_player = current_player 
            @winner = winner 
            @log = log
        end

        # Métodos Públicos
        
        def labyrinth
            @labyrinth
        end

        def players
            @players
        end

        def monsters
            @monsters
        end

        def current_player
            @currentPlayer
        end

        def winner
            @winner
        end

        def log
            @log
        end

        def to_s

            cadena = ""
            cadena << @labyrinth << "\n"
            cadena << @players << "\n"
            cadena << @monsters << "\n"
            cadena << "TURNO: JUGADOR " << @current_player.to_s << "\n"

            if (@winner) 
                cadena << "HAY GANADOR!!!!" << "\n" << "FELICIDADES JUGADOR " <<
                @current_player.to_s << "\n";
            end

            cadena << @log << "\n"

            cadena

        end

    end
end

File: Java/irrgarten/irrgarten/src/irrgarten/GameState.java

package irrgarten;

/**
 * @brief   Permite almacenar una representación del estado completo del juego
 * @author  Isabel Morro Tabares
 */

public class GameState {
    
    // Atributos de instancia privados
    
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;         // Guarda información relevante de la ronda anterior
    
    // Constructor
    
    public GameState (String _labyrinth, String _players, String _monsters,
                      int _currentPlayer, boolean _winner, String _log){
        
        labyrinth = _labyrinth;
        players = _players;
        monsters = _monsters;
        currentPlayer = _currentPlayer;
        winner = _winner;
        log = _log;
        
    }
    
    // Consultores
    
    public String getLabyrinth (){
        return labyrinth;
    }
    
    public String getPlayers (){
        return players;
    }
    
    public String getMonsters(){
        return monsters;
    }
    
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    
    public boolean isWinner(){
        return winner;
    }
    
    public String getLog(){
        return log;
    }
   
    @Override
    public String toString(){
        String cadena = "";
       
        cadena += labyrinth + "\n";
        cadena += players + "\n";
        cadena += monsters + "\n";
        cadena += "TURNO: JUGADOR " + currentPlayer + "\n";
        
        if (winner) cadena += "HAY GANADOR!!!!" + "\n" + "FELICIDADES JUGADOR " 
                    + currentPlayer + "\n";
        
        if (!winner) cadena += log + "\n";
        
        return cadena;
   }
}
